internal/wal_fetch_handler.go

```diff
@@ -149,16 +153,74 @@ func DecompressWALFile(dst io.Writer, archiveReader io.ReadCloser, decompressor
 	return err
 }
 
+// CachedDecompressor is the file extension describing decompressor
+type CachedDecompressor struct {
+	FileExtension string
+}
+
+func LastDecompressor() (compression.Decompressor, error) {
+	var cache CachedDecompressor
+	var cacheFilename string
+
+	usr, err := user.Current()
+	if err == nil {
+		cacheFilename = filepath.Join(usr.HomeDir, ".walg_decompressor_cache")
+		file, err := ioutil.ReadFile(cacheFilename)
+		if err == nil {
+			_ = json.Unmarshal(file, &cache)
+			return compression.FindDecompressor(cache.FileExtension), nil
+		}
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+func SetLastDecompressor(decompressor compression.Decompressor) {
+	var cache CachedDecompressor
+	usr, err := user.Current()
+
+	if err != nil {
+		return
+	}
+
+	cacheFilename := filepath.Join(usr.HomeDir, ".walg_decompressor_cache")
+	cache.FileExtension = decompressor.FileExtension()
+
+	marshal, err := json.Marshal(&cache)
+	if err == nil && len(cacheFilename) > 0 {
+		_ = ioutil.WriteFile(cacheFilename, marshal, 0644)
+	}
+}
+
+func tryCachedFormatFirst(decompressors []compression.Decompressor) []compression.Decompressor {
```

Comment:
Does this function just try to get cached decompressors? What format means here?